src/BugsnagUnity.mm

```diff
@@ -795,57 +764,33 @@ void bugsnag_startBugsnagWithConfiguration(const void *configuration, char *noti
   [BSG_KSCrash sharedInstance].introspectMemory = NO;
 }
 
-void bugsnag_addBreadcrumb(char *message, char *type, char *metadata[], int metadataCount) {
+void bugsnag_addBreadcrumb(char *message, char *type, char *metadataJson) {
   NSString *ns_message = [NSString stringWithUTF8String: message == NULL ? "<empty>" : message];
   [Bugsnag.client addBreadcrumbWithBlock:^(BugsnagBreadcrumb *crumb) {
       crumb.message = ns_message;
       crumb.type = BSGBreadcrumbTypeFromString([NSString stringWithUTF8String:type]);
-      if (metadataCount > 0) {
-        NSMutableDictionary *ns_metadata = [NSMutableDictionary new];
-
-        for (int i = 0; i < metadataCount - 1; i += 2) {
-          char *key = metadata[i];
-          char *value = metadata[i+1];
-          if (key == NULL || value == NULL)
-              continue;
-          NSString *ns_key = [NSString stringWithUTF8String:key];
-          NSString *ns_value = [NSString stringWithUTF8String:value];
-          ns_metadata[ns_key] = ns_value;
-        }
-
-        crumb.metadata = ns_metadata;
+      if (metadataJson != NULL) {
+        crumb.metadata = getDictionaryFromMetadataJson(metadataJson);
       }
   }];
 }
 
-void bugsnag_retrieveBreadcrumbs(const void *managedBreadcrumbs, void (*breadcrumb)(const void *instance, const char *name, const char *timestamp, const char *type, const char *keys[], int keys_size, const char *values[], int values_size)) {
+void bugsnag_retrieveBreadcrumbs(const void *managedBreadcrumbs, void (*breadcrumb)(const void *instance, const char *name, const char *timestamp, const char *type, const char *metadataJson)) {
   [Bugsnag.breadcrumbs enumerateObjectsUsingBlock:^(BugsnagBreadcrumb *crumb, __unused NSUInteger index, __unused BOOL *stop){
     const char *message = [crumb.message UTF8String];
     const char *timestamp = [[BSG_RFC3339DateTool stringFromDate:crumb.timestamp] UTF8String];
     const char *type = [BSGBreadcrumbTypeValue(crumb.type) UTF8String];
 
     NSDictionary *metadata = crumb.metadata;
-
-    NSArray *keys = [metadata allKeys];
-    NSArray *values = [metadata allValues];
-
-    int count = 0;
-
-    if ([keys count] <= INT_MAX) {
-      count = (int)[keys count];
+    if(metadata != NULL)
+    {
+        breadcrumb(managedBreadcrumbs, message, timestamp, type, getMetadataJson(metadata));
```

Comment:
Now it's safe to pass NULL to `getMetadataJson()` ...

```suggestion
    breadcrumb(managedBreadcrumbs, message, timestamp, type, getMetadataJson(metadata));
```

and no need for the else! 🎉

Reply:
Done